mysql-test/suite/rdrs/include/have_rdrs.inc

```diff
@@ -0,0 +1,199 @@
+use strict;
+
+use File::Basename;
+use IO::File;
+use lib "lib/";
+use My::Find;
+use IO::Select;
+use IO::Socket::INET;
+
+#
+# Look for rdrs library, if not found: skip test.
+#
+
+#
+# Set up paths
+#
+my $vardir = $ENV{MYSQLTEST_VARDIR} or die "Need MYSQLTEST_VARDIR";
+my $mysql_test_dir = $ENV{MYSQL_TEST_DIR} or die "Need MYSQL_TEST_DIR";
+my $basedir = dirname($mysql_test_dir);
+my $version = $ENV{MYSQL_SERVER_VERSION};
+
+# In the RPM case, binaries and libraries are installed in the
+# default system locations, instead of having our own private base
+# directory. And we install "usr/share/mysql-test". Moving up two
+# more directory relative to "mysql-test" gives us a usable base
+# directory for RPM installs.
+if (! -d "$basedir/sql" and ! -d "$basedir/bin" )
+{
+  $basedir= dirname($basedir);
+}
+
+my $bindir = $ENV{MTR_BINDIR} || $basedir; # Out of source set MTR_BINDIR
+
+#
+# Check if the needed libs are available
+#
+
+# The expected installation locations are lib, lib/mysql, lib64/mysql, lib/x86_64-linux-gnu
+
+my $F = IO::File->new("$vardir/tmp/have_rdrs_result.inc", 'w') or die;
+print $F "# Looking for rdrs version $version\n";
+
+
+#---------------------------- ndbclient -----------------------------------------
+
+my $ndbclient_lib= my_find_file($bindir,
+                                [ "library_output_directory",
+                                  "lib*/*",
+                                  "lib"
+                                ],
+                                [ "ndbclient.dll",
+                                  "libndbclient.so",
+                                  "libndbclient.dylib"
+                                ],
+                                NOT_REQUIRED);
+
+if ($ndbclient_lib) {
+  my $ndbclient_lib_dir = dirname($ndbclient_lib);
+  print $F "--let \$RDRS_NDBCLIENT_LIB= $ndbclient_lib_dir\n"; 
+  print $F "--echo Found RonDB client library\n"
+} else {
+  # Test runs only when ndb is compiled and enabled.
+  # If not able to find libndbclient, probably an issue with this script.
+  print $F "die Did not find ndbclient library\n";
+}
+
+#---------------------------- rdrclient -----------------------------------------
+
+my $rdrclient_lib = my_find_file($bindir,
+                                [ "library_output_directory",
+                                  "lib*/*",
+                                  "lib"
+                                ],
+                                [ "rdrclient.dll",
+                                  "librdrclient.so",
+                                  "librdrclient.dylib"
+                                ],
+                                NOT_REQUIRED);
+
+if ($rdrclient_lib) {
+   my $rdrclient_lib_dir = dirname($rdrclient_lib);
+   print $F "--let \$RDRS_CLIENT_LIB= $rdrclient_lib_dir\n";
+   print $F "--echo Found RonDB REST server client library\n"
+} else {
+  print $F "--let \$skip_test=1\n";
+  print $F "--let \$skip_reason=This test requires RonDB REST library\n";
```

Comment:
die call?

Reply:
We skip the test if rdrs client lib is not found